go/vt/vttablet/tabletmanager/vreplication/http_stream_mock.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2019 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+ * To be used instead of HttpTest in cases where a test needs to do synchronized sequential reads are required from the
+ * http stream. Note that this handles only one write at a time: the test should read the written data before the next
+ * write. Data is sent to the channel only on a Flush.
+ */
+
+package vreplication
+
+import (
+	"net/http"
+)
+
+//HTTPStreamWriterMock implements http.ResponseWriter and adds a channel to sync writes and reads
```

Comment:
I thought gofmt should have nit-picked this. There should be a space between `//` and the start of the comment. Same comments below.